pymeasure/instruments/etslindgren/m7006_001.py

```diff
@@ -47,127 +47,110 @@ def __init__(self, resource_name, slot=1, device="A", **kwargs):
-
-    def down(self):
-        """Move the mast down."""
-        return self.adapter.write(f"{self._slot}{self._device}DN")
-
-    def up(self):
-        """Move the mast up."""
-        return self.adapter.write(f"{self._slot}{self._device}UP")
-
-    def stop(self):
-        """Stop the device."""
-        return self.adapter.write(f"{self._slot}{self._device}ST")
-
-    @property
-    def polarity(self):
-        """Return the polarity of the antenna boom. returns either H or V."""
-        result = self.adapter.write(f"{self._slot}{self._device}P?")
-        if result == 1:
-            return "H"
-        elif result == 0:
-            return "V"
-        else:
-            raise ValueError("Polarity not recognized")
-
-    @polarity.setter
-    def polarity(self, value):
-        """Set the polarity of the antenna boom.
-
-        Value must be either H or V.
-        """
-        value = strict_discrete_set(value, "H", "V")
-        value = map()
-        return self.adapter.write(f"{self._slot}{self._device}P{value}")
-
-    @property
-    def position(self):
-        """Return the current position of the device."""
-        return self.adapter.ask(f"{self._slot}{self._device}CP?")
-
-    @position.setter
-    def position(self, value):
-        """Set the current position of the device.
-
+    def write(self, command):
+        super().write(f"{self._slot}{self._device}{command}")
+
+    def values(self, command, **kwargs):
+        return super().values(f"{self._slot}{self._device}{command}", **kwargs)
+
+    acceleration = Instrument.control(
+        "ACC?",
+        "ACC %g",
+        """Acceleration of the device in seconds. Settable 0.1 and 30,
+        and gettable.
+        """,
+        values=(0.1, 30),
+        validator=strict_range,
+    )
+
+    speed = Instrument.control(
+        "SPEED?",
+        "SPEED %g",
+        """Speed of the device, as a percentage of the max speed.
+        Settable 0.0 to 100.0, and gettable.""",
+        values=(0.0, 100.0),
+        validator=strict_range,
+    )
+
+    aux1 = Instrument.control(
+        "AUX1?",
+        "AUX1 %g",
+        """Auxiliary input 1, Settable to ON or OFF, and gettable.""",
+        values=("ON", "OFF"),
+        validator=strict_discrete_set,
+    )
+
+    aux2 = Instrument.control(
+        "AUX2?",
+        "AUX2 %g",
+        """Auxiliary input 2, Settable to ON or OFF, and gettable.""",
+        values=("ON", "OFF"),
+        validator=strict_discrete_set,
+    )
+
+    ccw = Instrument.setting(
+        "CCW",
+        """Move the TurnTable in the counter clockwise direction.""",
+    )
+
+    cw = Instrument.setting(
+        "CW",
+        """Move the TurnTable in the clockwise direction.""",
+    )
+
+    down = Instrument.setting(
+        "DN",
+        """Move the mast down.""",
+    )
+    up = Instrument.setting(
+        "UP",
+        """Move the mast up.""",
+    )
+    stop = Instrument.setting("ST", """Stop the device.""")
```

Comment:
A setting is called in the style `instr.cw = 5`. These settings, however, seem to set not a value, at least they do not have a parameter.
Would it be better to have a setting `instr.rotation = "cw"/"ccw"`? and `direction = up/down`? You can use `values={'down': "DN", 'up': "UP"}` and `map_values=True` in order to convert english words to the abbreviation.

Stop should be a method, I think, as you do not set anything or get anything.